templates/libvirt/devices/block.xml

```diff
@@ -2,20 +2,21 @@
     <driver name="phy" />
     <source dev="{{ device.device_node }}" />
     {%- if 'frontend-dev' in options %}
-        <target dev="{{ options.get('frontend-dev') }}" />
+    <target dev="{{ options.get('frontend-dev') }}" />
     {%- elif options.get('devtype', 'disk') == 'cdrom' and not vm.kernel %}
-        <!-- prefer xvdd for CDROM -->
-        <target dev="xvdd" />
+    <!-- prefer xvdd for CDROM -->
+    <target dev="xvdd" />
     {%- else %}
-        <target dev="xvd{{dd[counter.i]}}" />
+    <target dev="xvd{{dd[counter.i]}}" />
         {% if counter.update({'i': counter.i + 1}) %}{% endif %}
     {%- endif %}
 
     {%- if options.get('read-only', 'no') == 'yes' %}
-        <readonly />
+    <readonly />
     {%- endif %}
 
     {%- if device.backend_domain.name != 'dom0' %}
     <backenddomain name="{{ device.backend_domain.name }}" />
     {%- endif %}
+    <script path="/etc/xen/scripts/qubes-block" />
```

Comment:
There are a few issues with using this not-script for qvm-block:
1. It will remove the loop device on detach, even if it didn't created it in the first place (pre-existing loop dev).
2. It requires the backend to use template from R4.2 (in other words: breaks compatibility with templates migrated from older system but not upgraded to R4.2 yet).

The second one is not a big issue in practice - while we try to not break existing features with new releases within the same major version, I suspect block backend outside of dom0 is mostly used in sys-usb, and there is little motivation to migrate system qubes from older version (especially when it's disposable). That said, R4.2 finally [advertises qubes tools version inside template](https://github.com/QubesOS/qubes-core-admin-client/blob/main/doc/manpages/qvm-features.rst#qubes-agent-version), so you could put the above under `{% if device.backend_domain.check_with_template('qubes-agent-version', '4.1') >= '4.2' %}` (`>=` on strings wont work for `4.10`, but I don't think we'll reach such high versions). The dom0 backend case don't need this check (which means either duplicated script line, or longer condition).

But the first issue needs fixing if this line is to stay. You need some way to distinguish if loop device was created by the qubes-block. Maybe compare `physical-device-path` xenstore entry against `params`?

(this issue is the cause of all the failures in `system_tests_devices` job - it creates loop device, and then does qvm-block attach, qvm-block detach, qvm-block attach - the last attach fails)